Sources/NIO/CircularBuffer.swift

```diff
@@ -151,6 +243,18 @@ public struct CircularBuffer<E>: CustomStringConvertible, AppendableCollection {
         return (self.headIdx + index) & self.mask
     }
 
+    /// Returns the internal buffer next index after `index`.
+    private func bufferIndex(after: Int) -> Int {
+        let nextIndex = (after + 1) >= self.capacity ? 0 : (after + 1)
```

Comment:
Rather than use the ternary, maybe just use the masking math that we do elsewhere?